app/controllers/ViewAppealDetailsController.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import controllers.predicates.{AuthPredicate, DataRequiredAction, DataRetrievalAction}
+import helpers.SessionAnswersHelper
+import javax.inject.Inject
+import models.NormalMode
+import models.pages.{PageMode, ViewAppealDetailsPage}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.ViewAppealDetailsPage
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ViewAppealDetailsController @Inject()(viewAppealDetailsPage: ViewAppealDetailsPage,
+                                            sessionAnswersHelper: SessionAnswersHelper)
+                                           (implicit mcc: MessagesControllerComponents,
+                                            appConfig: AppConfig,
+                                            authorise: AuthPredicate,
+                                            dataRetrieval: DataRetrievalAction) extends FrontendController(mcc) with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = (authorise andThen dataRetrieval) {
```

Comment:
The data retrieval step will retrieve the appeal details using the current journeyId, which would change if the user starts a new appeal. This key should be used for this context:
https://github.com/hmrc/penalties-appeals-frontend/blob/1a5233d59655de6395ff044588e301bfa9e8dcec/app/utils/SessionKeys.scala#L65